lib/spack/spack/spec.py

```diff
@@ -1253,8 +1172,9 @@ def from_node_dict(node):
                 if name in _valid_compiler_flags:
                     spec.compiler_flags[name] = value
                 else:
-                    spec.variants[name] = VariantSpec(name, value)
-
+                    spec.variants[name] = VariantSpec.from_node_dict(
+                        name, value
+                    )
         elif 'variants' in node:
```

Comment:
This is for backwards compatibility. Specs from before #360 (cflags) used "variants." In #360 we opted to move to the more general "parameters" instead of adding a separate top-level designation for "flags"